packages/fedify/src/federation/kv.ts

```diff
@@ -148,4 +175,33 @@ export class MemoryKvStore implements KvStore {
     this.#values[encodedKey] = [newValue, expiration];
     return Promise.resolve(true);
   }
+
+  /**
+   * {@inheritDoc KvStore.list}
+   */
+  async *list(prefix?: KvKey): AsyncIterable<KvStoreListEntry> {
+    const now = Temporal.Now.instant();
+    for (const [encodedKey, entry] of Object.entries(this.#values)) {
+      const key = JSON.parse(encodedKey) as KvKey;
+      // Check prefix match
+      if (prefix != null) {
+        if (key.length < prefix.length) continue;
+        let matches = true;
+        for (let i = 0; i < prefix.length; i++) {
+          if (key[i] !== prefix[i]) {
+            matches = false;
+            break;
+          }
+        }
```

Comment:
This is good but how about:
```suggestion
        const matches = prefix.every((p, i) => p === key[i]);
```